### ISSUE TITLE
Indent for binary operators

### ISSUE THREAD

**Issue by golopot** [eslint/eslint#12427]
**What rule do you want to change?**
indent

**Does this change cause the rule to produce more or fewer warnings?**
more

**How will the change be implemented? (New option, new default behavior, etc.)?**
new option to check indent in expressions containing a binary operator.

**Please provide some example code that this change will affect:**

```js
if ( 
a &&
   b &&
       c
) {}

const b = 1 +
     2 +
3;

```

**What does the rule currently do for this code?**
No warn.

**What will the rule do after it's changed?**
Do warn.

**Are you willing to submit a pull request to implement this change?**
No.

**Comment by kaicataldo** [comment]
I believe this is a dupe of https://github.com/eslint/eslint/issues/12248, unless you're proposing we only cover the case of expressions with binary operators.

**Comment by golopot** [comment]
Yes it looks like so. Handling all multiline expressions will certainly handle this case.
